snuba/utils/streams/backends/local/backend.py

```diff
@@ -44,39 +44,33 @@ def __init__(self, clock: Clock = TestingClock()) -> None:
         # The active subscriptions are stored by consumer group as a mapping
         # between the consumer and it's subscribed topics.
         self.__subscriptions: MutableMapping[
-            str, MutableMapping[DummyConsumer[TPayload], Sequence[Topic]]
+            str, MutableMapping[LocalConsumer[TPayload], Sequence[Topic]]
         ] = defaultdict(dict)
 
         self.__lock = Lock()
 
     def get_consumer(
         self, group: str, enable_end_of_partition: bool = False
     ) -> Consumer[TPayload]:
-        return DummyConsumer(
+        return LocalConsumer(
             self, group, enable_end_of_partition=enable_end_of_partition
         )
 
     def get_producer(self) -> Producer[TPayload]:
-        return DummyProducer(self)
+        return LocalProducer(self)
```

Comment:
Very likely that these two methods get pulled out into a `Broker` (or `Cluster`?) that gets moved to `streams.abstract` in the near future and also gets used for instantiating Kafka consumers and producers associated with a cluster.

This is only really blocked on:

1. the `KafkaConsumerWithCommitLog`, which needs to be made Kafka-agnostic,
2. existing direct usages of the Confluent Kafka producer (including the commit log producer, above),
3. some updates to the consumer builder